### PR TITLE
feat: read static files from the public folder

### LINKED ISSUE #7: Support static assests in handlers
Some handlers return HTML code that may require static assets such as JavaScript (client-side) and CSS. Currently, it can be injected in the handler, however is much more convenient to provide a way to set static files.

We can provide this feature using two approaches:

* Mount generic folders (using WASI) in the handler and provide libraries to manage those files
* Add a new `data.folder` configuration property that will serve the files directly using the Rust server

The final approach is still under discussion.

### PR DESCRIPTION
**Pull request by Angelmmiguel**
Support static files by reading them from the `public` folder in the root of the project. Any file inside this folder will be served as a static file, so project like static sites can work without embedding files in the workers. This is a pretty consistent standard in different web frameworks like:

* [NextJS](https://nextjs.org/docs/basic-features/static-file-serving)
* [Astro](https://docs.astro.build/en/core-concepts/project-structure/#public)
* [Remix](https://remix.run/docs/en/v1/tutorials/jokes#explore-the-project-structure)
* etc

Note that the `public` folder based on the given path in the CLI. So, for `wws ./examples`, the CLI will serve the files from `./examples/public`. There's an issue with path management as `wws ./examples` should calculate the routes without including the `examples` folder. This issue will be solved in #3.

It closes #7

### CODE REVIEW

**Review by gzurl — APPROVED** [review]
LGTM!

**Comment by Angelmmiguel** [comment]
Thanks for the review @gzurl ! I'm going to add the missing use case of allowing `html` files in the `public` folder.

**Review by assambar — APPROVED** [review]
Ship it!

You could add this later, but I think a simple example worker that uses static assets will help the adoption of this feature.
IDK, for example something simple that creates a business card with some background image based on a call to `make-business-card.js?first_name=John&last_name=Doe&phone=123345...`

**Comment by Angelmmiguel** [comment]
@gzurl @assambar @ereslibre I completed the feature by adding support for static HTML files. Now, the `/public/about.html` file will be mounted in `/about` automatically.

> You could add this later, but I think a simple example worker that uses static assets will help the adoption of this feature.
IDK, for example something simple that creates a business card with some background image based on a call to make-business-card.js?first_name=John&last_name=Doe&phone=123345...

This is an interesting use case. It can work for OpenGraph images and business cards (as you suggested). I will give a second though to that idea and will implement some examples around. However, those will work purely with workers, so I don't think static assets will be used that much in the example.

File: src/main.rs [hunk 1]
```diff
@@ -169,6 +192,51 @@ async fn main() -> std::io::Result<()> {
...
+                        Path::new(base_path).join(format!("public{}.html", uri_path));
+                    let public_404_path = Path::new(base_path).join("public").join("404.html");
+
+                    if uri_path.ends_with("/") && index_folder_path.exists() {
+                        file = NamedFile::open_async(index_folder_path).await;
+                    } else if !uri_path.ends_with("/") && html_ext_path.exists() {
+                        file = NamedFile::open_async(html_ext_path).await;
+                    } else {
+                        file = NamedFile::open_async(public_404_path).await;
+                    }
```
Comment (ereslibre) [inline src/main.rs:225]: I think this logic could be extracted to a couple of functions, just to keep this closure a bit more readable.

**Review by ereslibre — APPROVED** [review]
Looking fantastic; thanks @Angelmmiguel! LGTM. A minor optional comment.

File: src/main.rs (hunk 1 above)
Comment (Angelmmiguel) [reply src/main.rs]: I extracted this logic into a separate method. Thanks for the suggestion!

**Pull request merged**